heron/api/src/java/org/apache/heron/streamlet/StreamGrouper.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.heron.streamlet;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.heron.api.grouping.StreamGrouping;
+import org.apache.heron.api.utils.Utils;
+
+/**
+ * StreamGrouper is the class to config grouping strategy in Streamlet API. Note that
+ * One Operator could have more than one grouping strategies (when there are more than one
+ * incoming streams) although it has one in most cases.
+ */
+public class StreamGrouper {
+  /*
+   * Map of stream id to grouping strategies. One operator can have more than one incoming streams
+   * and each stream needs a grouping strategies. Hence it is possible for an operator to have
+   * more than one entries.
+   */
+  private HashMap<String, StreamGrouping> groupingMap;
+
+  /**
+   * Construct a Grouper object with specific grouping strategy and default stream id
+   * @param grouping The grouping strategy to be applied
+   */
+  public StreamGrouper(StreamGrouping grouping) {
+    groupingMap = new HashMap<>();
+    append(Utils.DEFAULT_STREAM_ID, grouping);
+  }
+
+  /**
+   * Construct a Grouper object with specific grouping strategy and streamid
+   * @param streamId The stream id that the strategy to be applied to
+   * @param grouping The grouping strategy to be applied
+   */
+  public StreamGrouper(String streamId, StreamGrouping grouping) {
+    groupingMap = new HashMap<>();
```

Comment:
- `HashMap` accepts `null` K,V pairs so `ConcurrentHashMap` can be preferred to avoid `null` access (due to Public API)
- On the other hand, `streamId` also needs to be robust for blank values(`streamId` = " ")

Reply:
Good to know. Thx. Will update.